soh/soh/Enhancements/randomizer/3drando/location_access/locacc_ganons_castle.cpp

```diff
@@ -376,4 +369,54 @@ void RegionTable_Init_GanonsCastle() {
                   EventAccess(&logic->LightTrialClear, {[]{return logic->CanUse(RG_LIGHT_ARROWS);}}),
   }, {}, {});
   }
+
+  /*--------------------------
+  |     TOWER AND ESCAPE     |
+  ---------------------------*/
+  areaTable[RR_GANONS_TOWER_FLOOR_1] = Region("Ganon's Tower Floor 1", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {}, {
+                  //Exits
+                  Entrance(RR_GANONS_CASTLE_LOBBY,  {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_FLOOR_2, {[]{return logic->CanKillEnemy(RE_DINOLFOS, ED_CLOSE, true, 2);}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_FLOOR_2] = Region("Ganon's Tower Floor 2", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  LOCATION(RC_GANONS_TOWER_BOSS_KEY_CHEST, logic->CanKillEnemy(RE_STALFOS, ED_CLOSE, true, 2)),
+  }, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_1, {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_FLOOR_3, {[]{return logic->CanKillEnemy(RE_STALFOS, ED_CLOSE, true, 2);}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_FLOOR_3] = Region("Ganon's Tower Floor 3", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {}, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_2,        {[]{return true;}}),
+                  Entrance(RR_GANONS_TOWER_GANONDORF_LAIR, {[]{return logic->CanKillEnemy(RE_IRON_KNUCKLE, ED_CLOSE, true, 2) && logic->HasItem(RG_GANONS_CASTLE_BOSS_KEY);}}),
+  });
+
+  areaTable[RR_GANONS_TOWER_GANONDORF_LAIR] = Region("Ganondorf's Lair", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  LOCATION(RC_GANONDORF_HINT, logic->HasBossSoul(RG_GANON_SOUL)),
+                  //18 pots
+  }, {
+                  //Exits
+                  Entrance(RR_GANONS_TOWER_FLOOR_3, {[]{return true;}}),
+                  Entrance(RR_GANONS_CASTLE_ESCAPE, {[]{return logic->CanKillEnemy(RE_GANONDORF);}}),
+  });
+
+  areaTable[RR_GANONS_CASTLE_ESCAPE] = Region("Ganon's Castle Escape", "Ganons Castle", {RA_GANONS_CASTLE}, NO_DAY_NIGHT_CYCLE, {}, {
+                  //Locations
+                  //10 pots
+  }, {
+                  //Exits
+                  //temporary
+                  Entrance(RR_GANONS_CASTLE_GANON_ARENA, {[]{return true;}}),
+                  //real logic once we figure out how to deal with castle escape skip
+                  //Entrance(RR_GANONS_CASTLE_GANON_ARENA, {[]{return logic->CanKillEnemy(RE_STALFOS, ED_CLOSE, true, 2, true) && logic->CanAvoidEnemy(RE_GIBDO);}}),
```

Comment:
Assuming the "gibdo" is the redead on the bridge near the end, there's enough room to get around it without a logic check. 

CanAvoid is meant to be for situations where a chasing enemy can interrupt link in the middle of something(think lab roof), not for if the enemy physically blocks Links path but is bad at it, if there's room to go around like this you just don't need to make an entry at all.

Reply:
The gibdo instead of redead part was a brainfart.
I wasn't sure if I needed to use `CanAvoidEnemy` but I thought that it was best to be safe.